libraries/APM_Control/AP_RollController.cpp

```diff
@@ -202,13 +202,26 @@ float AP_RollController::get_servo_out(int32_t angle_err, float scaler, bool dis
     angle_err_deg = angle_err * 0.01;
     float desired_rate = angle_err_deg/ gains.tau;
 
+    /*
+      prevent indecision in the roll controller when target roll is
+      close to 180 degrees from the current roll
+     */
+    const float indecision_threshold_deg = 160;
+    if (fabsf(angle_err_deg) > indecision_threshold_deg) {
+        if (desired_rate * last_desired_rate < 0) {
+            desired_rate = -desired_rate;
```

Comment:
I think this should recalculate the angle error. The problem is as the error passes 180 it will start getting smaller again while still trying to go the long way round. We can make it continue to get larger until it flips at 160.

I think something like this would work, but I have not tested. 
```
            desired_rate =  (360 - fabsf(angle_err_deg)) / gains.tau * is_positive(angle_err_deg) ? 1.0 : -1.0;
```

Reply:
fixed with a different expression that I think is clearer